protocol.go

```diff
@@ -201,41 +189,46 @@ func (p *Protocol) clearWrites(start, end uint16) {
 	emptyBufferIndices(second)
 }
 
-func (p *Protocol) ReadPacket(header PacketHeader, buf []byte) []byte {
+func (p *Protocol) ReadPacket(header PacketHeader, buf []byte) (needed bool) {
```

Comment:
warning: exported method Protocol.ReadPacket should have comment or be unexported (golint)